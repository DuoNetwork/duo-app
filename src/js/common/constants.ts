export const AC_BUY_SELL = 'buySell';
export const AC_REFRESH = 'refresh';
export const AC_HISTORY = 'history';
