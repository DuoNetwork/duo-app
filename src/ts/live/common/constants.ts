export * from '../../../../../duo-admin/src/constants';

export const AC_CTD_STATES = 'custodianStates';
export const AC_CTD_PRICES = 'custodianPrices';
export const AC_ACCOUNT = 'account';
export const AC_BALANCES = 'balances';
export const AC_ADDRESSES = 'addresses';
export const AC_DNM_STATUS = 'dynamoStatus';
export const AC_DMN_HOURLY = 'dynamoHourly';
export const AC_DMN_MINUTELY = 'dynamoMinutely';
export const AC_DMN_PRICES = 'dynamoPrices';
export const AC_REFRESH = 'refresh';
