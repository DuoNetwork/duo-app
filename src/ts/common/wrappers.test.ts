import * as Constants from '@finbook/duo-contract-wrapper/dist/constants';
import { kovan } from '@finbook/duo-contract-wrapper/dist/contractAddresses';
jest.mock('@finbook/duo-contract-wrapper', () => ({
	Constants: Constants,
	Web3Wrapper: jest.fn(() => ({ contractAddresses: kovan })),
	DualClassWrapper: jest.fn(() => ({
		contract: 'dualClassWrapper'
	})),
	EsplanadeWrapper: jest.fn(() => ({
		contract: 'EsplanadeWrapper'
	})),
	MagiWrapper: jest.fn(() => ({
		contract: 'MagiWrapper'
	}))
}));

jest.mock('@finbook/duo-contract-wrapper/dist/StakeContractWrapper', () => ({
	StakeContractWrapper: jest.fn(() => ({
		contract: 'StakeContractWrapper'
	}))
}));

import {
	DualClassWrapper,
	EsplanadeWrapper,
	MagiWrapper,
	Web3Wrapper
} from '@finbook/duo-contract-wrapper';
import { StakeContractWrapper } from '@finbook/duo-contract-wrapper/dist/StakeContractWrapper';

import {
	dualClassWrappers,
	esplanadeWrapper,
	magiWrapper,
	stakeWrapper,
	web3Wrapper
} from './wrappers';

test('Web3Wrapper', () => {
	expect(web3Wrapper).toBeTruthy();
	expect((Web3Wrapper as any).mock.calls).toMatchSnapshot();
});

test('DualClassWrapper', () => {
	expect(dualClassWrappers).toMatchSnapshot();
	expect((DualClassWrapper as any).mock.calls).toMatchSnapshot();
});

test('EsplanadeWrapper', () => {
	expect(esplanadeWrapper).toMatchSnapshot();
	expect((EsplanadeWrapper as any).mock.calls).toMatchSnapshot();
});

test('MagiWrapper', () => {
	expect(magiWrapper).toMatchSnapshot();
	expect((MagiWrapper as any).mock.calls).toMatchSnapshot();
});

test('StakeContractWrapper', () => {
	expect(stakeWrapper).toMatchSnapshot();
	expect((StakeContractWrapper as any).mock.calls).toMatchSnapshot();
});
