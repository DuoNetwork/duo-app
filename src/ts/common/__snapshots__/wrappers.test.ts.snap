// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DualClassWrapper 1`] = `
Object {
  "Beethoven": Object {
    "Perpetual": Object {
      "contract": "dualClassWrapper",
    },
  },
  "Mozart": Object {
    "Perpetual": Object {
      "contract": "dualClassWrapper",
    },
  },
  "Vivaldi": Object {
    "100C-3H": Object {
      "contract": "dualClassWrapper",
    },
  },
}
`;

exports[`DualClassWrapper 2`] = `
Array [
  Array [
    Object {
      "contractAddresses": Object {
        "Custodians": Object {
          "Beethoven": Object {
            "Perpetual": Object {
              "aToken": Object {
                "address": "0x59E6B3d43F762310626d2905148939973db2BBd3",
                "code": "aETH",
              },
              "bToken": Object {
                "address": "0x8A3bEca74E0E737460bDE45a09594A8D7D8c9886",
                "code": "bETH",
              },
              "custodian": Object {
                "address": "0x13016f27945f3f7b39A5Daae068D698e34E55491",
                "code": "BEETHOVEN-PPT",
              },
            },
          },
          "Mozart": Object {
            "Perpetual": Object {
              "aToken": Object {
                "address": "0x2cB7CDDF82AD7fF2b207AD43586976FEB19BA985",
                "code": "sETH",
              },
              "bToken": Object {
                "address": "0x977feC72C295EdAf7E7deA5681E42CaFf69749e3",
                "code": "LETH",
              },
              "custodian": Object {
                "address": "0x00be45Fe5903AB1b33a9d3969b05b29552a6d18b",
                "code": "MOZART-PPT",
              },
            },
          },
          "Vivaldi": Object {
            "100C-3H": Object {
              "aToken": Object {
                "address": "0x2C4bD22588af822a95Fb9D7e972A1F1C4aE28F13",
                "code": "ETH-100C-3H",
              },
              "bToken": Object {
                "address": "0x9f49bb024EB176d227d4b85A58730a5dDeEF529D",
                "code": "ETH-100P-3H",
              },
              "custodian": Object {
                "address": "0x5734Fe05187857066Ed00d963D4bBBeB619797f4",
                "code": "VIVALDI-100C-3H",
              },
            },
          },
        },
        "MultiSigManagers": Array [
          Object {
            "address": "0xD728681490d63582047A6Cd2fC80B1343C6AbA20",
            "code": "Esplanade",
          },
        ],
        "Oracles": Array [
          Object {
            "address": "0x0d729B3C11b3E6Bf5792d36f640f3Be6f187Dd67",
            "code": "Magi",
          },
        ],
      },
    },
    "0x13016f27945f3f7b39A5Daae068D698e34E55491",
  ],
  Array [
    Object {
      "contractAddresses": Object {
        "Custodians": Object {
          "Beethoven": Object {
            "Perpetual": Object {
              "aToken": Object {
                "address": "0x59E6B3d43F762310626d2905148939973db2BBd3",
                "code": "aETH",
              },
              "bToken": Object {
                "address": "0x8A3bEca74E0E737460bDE45a09594A8D7D8c9886",
                "code": "bETH",
              },
              "custodian": Object {
                "address": "0x13016f27945f3f7b39A5Daae068D698e34E55491",
                "code": "BEETHOVEN-PPT",
              },
            },
          },
          "Mozart": Object {
            "Perpetual": Object {
              "aToken": Object {
                "address": "0x2cB7CDDF82AD7fF2b207AD43586976FEB19BA985",
                "code": "sETH",
              },
              "bToken": Object {
                "address": "0x977feC72C295EdAf7E7deA5681E42CaFf69749e3",
                "code": "LETH",
              },
              "custodian": Object {
                "address": "0x00be45Fe5903AB1b33a9d3969b05b29552a6d18b",
                "code": "MOZART-PPT",
              },
            },
          },
          "Vivaldi": Object {
            "100C-3H": Object {
              "aToken": Object {
                "address": "0x2C4bD22588af822a95Fb9D7e972A1F1C4aE28F13",
                "code": "ETH-100C-3H",
              },
              "bToken": Object {
                "address": "0x9f49bb024EB176d227d4b85A58730a5dDeEF529D",
                "code": "ETH-100P-3H",
              },
              "custodian": Object {
                "address": "0x5734Fe05187857066Ed00d963D4bBBeB619797f4",
                "code": "VIVALDI-100C-3H",
              },
            },
          },
        },
        "MultiSigManagers": Array [
          Object {
            "address": "0xD728681490d63582047A6Cd2fC80B1343C6AbA20",
            "code": "Esplanade",
          },
        ],
        "Oracles": Array [
          Object {
            "address": "0x0d729B3C11b3E6Bf5792d36f640f3Be6f187Dd67",
            "code": "Magi",
          },
        ],
      },
    },
    "0x00be45Fe5903AB1b33a9d3969b05b29552a6d18b",
  ],
  Array [
    Object {
      "contractAddresses": Object {
        "Custodians": Object {
          "Beethoven": Object {
            "Perpetual": Object {
              "aToken": Object {
                "address": "0x59E6B3d43F762310626d2905148939973db2BBd3",
                "code": "aETH",
              },
              "bToken": Object {
                "address": "0x8A3bEca74E0E737460bDE45a09594A8D7D8c9886",
                "code": "bETH",
              },
              "custodian": Object {
                "address": "0x13016f27945f3f7b39A5Daae068D698e34E55491",
                "code": "BEETHOVEN-PPT",
              },
            },
          },
          "Mozart": Object {
            "Perpetual": Object {
              "aToken": Object {
                "address": "0x2cB7CDDF82AD7fF2b207AD43586976FEB19BA985",
                "code": "sETH",
              },
              "bToken": Object {
                "address": "0x977feC72C295EdAf7E7deA5681E42CaFf69749e3",
                "code": "LETH",
              },
              "custodian": Object {
                "address": "0x00be45Fe5903AB1b33a9d3969b05b29552a6d18b",
                "code": "MOZART-PPT",
              },
            },
          },
          "Vivaldi": Object {
            "100C-3H": Object {
              "aToken": Object {
                "address": "0x2C4bD22588af822a95Fb9D7e972A1F1C4aE28F13",
                "code": "ETH-100C-3H",
              },
              "bToken": Object {
                "address": "0x9f49bb024EB176d227d4b85A58730a5dDeEF529D",
                "code": "ETH-100P-3H",
              },
              "custodian": Object {
                "address": "0x5734Fe05187857066Ed00d963D4bBBeB619797f4",
                "code": "VIVALDI-100C-3H",
              },
            },
          },
        },
        "MultiSigManagers": Array [
          Object {
            "address": "0xD728681490d63582047A6Cd2fC80B1343C6AbA20",
            "code": "Esplanade",
          },
        ],
        "Oracles": Array [
          Object {
            "address": "0x0d729B3C11b3E6Bf5792d36f640f3Be6f187Dd67",
            "code": "Magi",
          },
        ],
      },
    },
    "0x5734Fe05187857066Ed00d963D4bBBeB619797f4",
  ],
]
`;

exports[`EsplanadeWrapper 1`] = `
Object {
  "contract": "EsplanadeWrapper",
}
`;

exports[`EsplanadeWrapper 2`] = `
Array [
  Array [
    Object {
      "contractAddresses": Object {
        "Custodians": Object {
          "Beethoven": Object {
            "Perpetual": Object {
              "aToken": Object {
                "address": "0x59E6B3d43F762310626d2905148939973db2BBd3",
                "code": "aETH",
              },
              "bToken": Object {
                "address": "0x8A3bEca74E0E737460bDE45a09594A8D7D8c9886",
                "code": "bETH",
              },
              "custodian": Object {
                "address": "0x13016f27945f3f7b39A5Daae068D698e34E55491",
                "code": "BEETHOVEN-PPT",
              },
            },
          },
          "Mozart": Object {
            "Perpetual": Object {
              "aToken": Object {
                "address": "0x2cB7CDDF82AD7fF2b207AD43586976FEB19BA985",
                "code": "sETH",
              },
              "bToken": Object {
                "address": "0x977feC72C295EdAf7E7deA5681E42CaFf69749e3",
                "code": "LETH",
              },
              "custodian": Object {
                "address": "0x00be45Fe5903AB1b33a9d3969b05b29552a6d18b",
                "code": "MOZART-PPT",
              },
            },
          },
          "Vivaldi": Object {
            "100C-3H": Object {
              "aToken": Object {
                "address": "0x2C4bD22588af822a95Fb9D7e972A1F1C4aE28F13",
                "code": "ETH-100C-3H",
              },
              "bToken": Object {
                "address": "0x9f49bb024EB176d227d4b85A58730a5dDeEF529D",
                "code": "ETH-100P-3H",
              },
              "custodian": Object {
                "address": "0x5734Fe05187857066Ed00d963D4bBBeB619797f4",
                "code": "VIVALDI-100C-3H",
              },
            },
          },
        },
        "MultiSigManagers": Array [
          Object {
            "address": "0xD728681490d63582047A6Cd2fC80B1343C6AbA20",
            "code": "Esplanade",
          },
        ],
        "Oracles": Array [
          Object {
            "address": "0x0d729B3C11b3E6Bf5792d36f640f3Be6f187Dd67",
            "code": "Magi",
          },
        ],
      },
    },
    "0xD728681490d63582047A6Cd2fC80B1343C6AbA20",
  ],
]
`;

exports[`MagiWrapper 1`] = `
Object {
  "contract": "MagiWrapper",
}
`;

exports[`MagiWrapper 2`] = `
Array [
  Array [
    Object {
      "contractAddresses": Object {
        "Custodians": Object {
          "Beethoven": Object {
            "Perpetual": Object {
              "aToken": Object {
                "address": "0x59E6B3d43F762310626d2905148939973db2BBd3",
                "code": "aETH",
              },
              "bToken": Object {
                "address": "0x8A3bEca74E0E737460bDE45a09594A8D7D8c9886",
                "code": "bETH",
              },
              "custodian": Object {
                "address": "0x13016f27945f3f7b39A5Daae068D698e34E55491",
                "code": "BEETHOVEN-PPT",
              },
            },
          },
          "Mozart": Object {
            "Perpetual": Object {
              "aToken": Object {
                "address": "0x2cB7CDDF82AD7fF2b207AD43586976FEB19BA985",
                "code": "sETH",
              },
              "bToken": Object {
                "address": "0x977feC72C295EdAf7E7deA5681E42CaFf69749e3",
                "code": "LETH",
              },
              "custodian": Object {
                "address": "0x00be45Fe5903AB1b33a9d3969b05b29552a6d18b",
                "code": "MOZART-PPT",
              },
            },
          },
          "Vivaldi": Object {
            "100C-3H": Object {
              "aToken": Object {
                "address": "0x2C4bD22588af822a95Fb9D7e972A1F1C4aE28F13",
                "code": "ETH-100C-3H",
              },
              "bToken": Object {
                "address": "0x9f49bb024EB176d227d4b85A58730a5dDeEF529D",
                "code": "ETH-100P-3H",
              },
              "custodian": Object {
                "address": "0x5734Fe05187857066Ed00d963D4bBBeB619797f4",
                "code": "VIVALDI-100C-3H",
              },
            },
          },
        },
        "MultiSigManagers": Array [
          Object {
            "address": "0xD728681490d63582047A6Cd2fC80B1343C6AbA20",
            "code": "Esplanade",
          },
        ],
        "Oracles": Array [
          Object {
            "address": "0x0d729B3C11b3E6Bf5792d36f640f3Be6f187Dd67",
            "code": "Magi",
          },
        ],
      },
    },
    "0x0d729B3C11b3E6Bf5792d36f640f3Be6f187Dd67",
  ],
]
`;

exports[`Web3Wrapper 1`] = `
Array [
  Array [
    [Window],
    "https://kovan.infura.io",
    "",
    false,
  ],
]
`;
